src/padding_mode.cpp

```diff
@@ -33,27 +33,20 @@ using namespace openssl;
 using SSL_RSA_OAEP_LABEL_Ptr = std::unique_ptr<uint8_t, SSLFree<uint8_t>>;
 
 /*
- * Helper
+ * Interface Destructors
  */
 
-class PrepareOpenSSLContextVisitor {
-public:
-    PrepareOpenSSLContextVisitor(openssl::SSL_EVP_PKEY_CTX_Ptr& ctx) : ctx(ctx) {}
-
-    void operator()(const MGF1& m) const {
-        m.prepareOpenSSLContext(ctx);
-    }
-
-    openssl::SSL_EVP_PKEY_CTX_Ptr& ctx;
-};
+RSAEncryptionPadding::~RSAEncryptionPadding() = default;
+RSASignaturePadding::~RSASignaturePadding() = default;
 
 /*
  * NoPadding
  */
 
-int NoPadding::getDataBlockSize(const AsymmetricPublicKey &key) const
+bool NoPadding::checkMessageSize(const AsymmetricPublicKey &key, size_t messageSize) const
 {
-    return key.getKeySize()/8;
+    size_t keySizeInBytes = key.getKeySize()/8;
+    return messageSize == keySizeInBytes;
```

Comment:
Is this enforced by OpenSSL? There is no mathematical reason for this? messageSize < keySizeInBytes should be just fine. In particular, if the size in bytes matches but the message is one bit too long, data ist lost?

Reply:
Yes this is enforced by openssl:
```
flen must not be more than RSA_size(rsa) - 11 for the PKCS #1 v1.5 based padding 
modes, not more than RSA_size(rsa) - 42 for RSA_PKCS1_OAEP_PADDING and
exactly RSA_size(rsa) for RSA_NO_PADDING.
```
See https://www.openssl.org/docs/man1.1.1/man3/RSA_public_encrypt.html

Message size shoud imho always be a multiple of 8 Bit in this lib because the lib only offers to en-/decrypt vector<uint8_t>. But indeed, it is possible that the key size is not a multiple of 8 bits. (Will fix this)...